torch/optim/lr_scheduler.py

```diff
@@ -154,8 +154,12 @@ def __init__(self, optimizer, step_size, gamma=0.1, last_epoch=-1):
         super(StepLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
-        return [base_lr * self.gamma ** (self.last_epoch // self.step_size)
-                for base_lr in self.base_lrs]
+        if (self.last_epoch == 0) or (self.last_epoch % self.step_size != 0):
```

Comment:
The special case for `last_epoch == 0` is a real stick up my craw. The problem here is an existing infelicity in LR design, also noted by a user in https://github.com/pytorch/pytorch/issues/8837

Unfortunately, fixing this would be an even bigger BC-breaking change. Nothing you have to do for this PR, but something good to keep in mind.

CC @soumith